apps/emqx_authn/src/enhanced_authn/emqx_enhanced_authn_scram_mnesia.erl

```diff
@@ -169,84 +177,80 @@ authenticate(
 authenticate(_Credential, _State) ->
     ignore.
 
-destroy(#{user_group := UserGroup}) ->
+destroy(State) ->
+    trans(fun ?MODULE:do_destroy/1, [State]).
+
+do_destroy(#{user_group := UserGroup}) ->
```

Comment:
pattern match in destroy/1?
otherwise the original `function_clause` exception will become a mnesia abort.